core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1888,6 +1922,18 @@ private[spark] class DAGScheduler(
     clearCacheLocs()
   }
 
+  /**
+   * Remove shuffle data mapping when node is decomissioned.
+   *
+   * @param host host of the node that is decommissioned
+   */
+  private[scheduler] def handleNodeDecommissioned(host: String) {
```

Comment:
So if we remove the shuffle files on decommissioning then we're essentially forcing a recompute even if the files are still present and the node hasn't gone away which doesn't seem ideal.

Reply:
We have tried to answer this query in this comment https://github.com/apache/spark/pull/27636#issuecomment-593338415